src/plugins/data_source_management/public/components/create_data_source_wizard/components/create_form/create_data_source_form.test.tsx

```diff
@@ -222,3 +237,129 @@ describe('Datasource Management: Create Datasource form', () => {
     expect(component.find(passwordIdentifier).first().props().isInvalid).toBe(false);
   });
 });
+
+describe('Datasource Management: Create Datasource form with different authType configurations', () => {
+  let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
+  const mockSubmitHandler = jest.fn();
+  const mockTestConnectionHandler = jest.fn();
+  const mockCancelHandler = jest.fn();
+
+  /* Scenario 1: Should render the page normally with all authMethod combinations */
+  test('should render normally with all authMethod combinations', () => {
+    const authMethodCombinationsToBeTested = [
+      [noAuthCredentialAuthMethod],
+      [usernamePasswordAuthMethod],
+      [sigV4AuthMethod],
+      [noAuthCredentialAuthMethod, usernamePasswordAuthMethod],
+      [noAuthCredentialAuthMethod, sigV4AuthMethod],
+      [usernamePasswordAuthMethod, sigV4AuthMethod],
+      [noAuthCredentialAuthMethod, usernamePasswordAuthMethod, sigV4AuthMethod],
+    ];
+
+    authMethodCombinationsToBeTested.forEach((authMethodCombination) => {
+      const mockedContext = mockManagementPlugin.createDataSourceManagementContext();
+      mockedContext.authenticationMethodRegistery = new AuthenticationMethodRegistery();
+
+      authMethodCombination.forEach((authMethod) => {
+        mockedContext.authenticationMethodRegistery.registerAuthenticationMethod(authMethod);
+      });
+
+      component = mount(
+        wrapWithIntl(
+          <CreateDataSourceForm
+            handleTestConnection={mockTestConnectionHandler}
+            handleSubmit={mockSubmitHandler}
+            handleCancel={mockCancelHandler}
+            existingDatasourceNamesList={['dup20']}
+          />
+        ),
+        {
+          wrappingComponent: OpenSearchDashboardsContextProvider,
+          wrappingComponentProps: {
+            services: mockedContext,
+          },
+        }
+      );
+
+      const testConnBtn = component.find(testConnectionButtonIdentifier).last();
+      expect(testConnBtn.prop('disabled')).toBe(true);
+    });
+  });
+
+  /* Scenario 2: options selector should be disabled when only one authMethod supported */
```

Comment:
do we cover the case that options selector should be disabled when none authMethod supported

Reply:
Thanks for the comment.

Covered here:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/5916/commits/c2da4e2ac9f5231e09ce37409e1995bfde5a4304#diff-1cadbf80d14e8c1df75fe7485f5e0dbb1157965ef1aa41c7ce22264b91c5a195R291